x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/ValuesDoubleAggregator.java

```diff
@@ -106,34 +118,143 @@ public void close() {
+                int count = -selectedCounts[group];
+                selectedCounts[group] = total;
+                total += count;
+            }
+
+            /*
+             * Build a list of ids to insert in order *and* convert the running
+             * count in selectedCounts[group] into the end index (exclusive) in
+             * ids for each group.
+             * Here we use the negative counts to signal that a group hasn't been
+             * selected and the id containing values for that group is ignored.
+             *
+             * For example, if
+             * | Group | Value Count | Selected |
+             * |-------|-------------|----------|
+             * |     0 | 3           | <-       |
+             * |     1 | 1           | <-       |
+             * |     2 | 2           |          |
+             * |     3 | 1           | <-       |
+             * |     4 | 4           | <-       |
+             *
+             * Then the total is 9 and the counts array will start with 0, 3, -2, 4, 5.
+             * The counts will end with 3, 4, -2, 5, 9.
+             */
+            reserveBytesForIntArray(total);
+
+            this.ids = new int[total];
+            for (int id = 0; id < hashes.size(); id++) {
+                int group = (int) hashes.getKey1(id);
+                ids[selectedCounts[group]++] = id;
+            }
+        }
+
         @Override
         public void close() {
-            releasable.close();
+            Releasables.closeExpectNoException(hashes, () -> blockFactory.adjustBreaker(-extraMemoryUsed));
         }
     }
 
     /**
      * State for a grouped {@code VALUES} aggregation. This implementation
-     * emphasizes collect-time performance over the performance of rendering
-     * results. That's good, but it's a pretty intensive emphasis, requiring
-     * an {@code O(n^2)} operation for collection to support a {@code O(1)}
-     * collector operation. But at least it's fairly simple.
+     * emphasizes collect-time performance over result rendering performance.
+     * The first value in each group is collected in the {@code firstValues}
+     * array, and subsequent values for each group are collected in {@code nextValues}.
      */
     public static class GroupingState implements GroupingAggregatorState {
-        private int maxGroupId = -1;
         private final BlockFactory blockFactory;
-        private final LongLongHash values;
+        DoubleArray firstValues;
+        private BitArray seen;
+        private int maxGroupId = -1;
+        private final NextValues nextValues;
 
         private GroupingState(DriverContext driverContext) {
-            this.blockFactory = driverContext.blockFactory();
-            values = new LongLongHash(1, driverContext.bigArrays());
+            DoubleArray _firstValues = null;
+            NextValues _nextValues = null;
+            try {
+                _firstValues = driverContext.bigArrays().newDoubleArray(1, false);
```

Comment:
I tend not to do the `_` thing in the ctor because this shadowing thing is so so common.

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/130510/commits/e390d42e2fd7f935e8f8f46c61e7b28ad0149c7b